build.override.targets

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0"  xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--if ServiceUri is set, the service has been already started on another machine-->
+  <Target Name="WcfSetup" Condition="($(WithCategories.Contains('OuterLoop')) or '$(OuterLoop)' == 'true') and ('$(ServiceUri)' == '') and $(OS.Contains('Windows'))" BeforeTargets="BuildAllProjects">
+    <Exec Command="$(MSBuildThisFileDirectory)\src\System.Private.ServiceModel\tools\scripts\WCFCustomSetup.cmd" ConsoleToMSBuild="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="WcfSetupOutput" />
+    </Exec>
+    <Message Text="WcfSetupOutput: $(WcfSetupOutput)" />
+  </Target>
+
```

Comment:
For the cleanup you seem to hook AfterTargets=Test but for the setup you did BeforeTargets=BuildAllTargets. I suspect you want to try and scope these onto the same target, otherwise they might not be in-sync.